### PR TITLE
Improve ClusterServiceVersion schema

### LINKED ISSUE #167: Improve API documentation for CSV and PackageManifest
The API docs for CSV and PackageManifests are incomplete for the following properties:

CSV properties:
```
.spec.customresourcedefinitions.owned[].resources[]
.spec.customresourcedefinitions.required[].resources[]
.spec.apiservicedefinitions.owned[].resources[]
.spec.apiservicedefinitions.required[].resources[]
```

These all share the same description:
> APIResourceReference is a Kubernetes resource type used by a custom resource

I've searched through OpenShift, Kubernetes, and Operator Framework repos and documentation and cannot find any further documentation of an `APIResourceReference` schema.  The schema consists of three string-type properties: `kind`,  ` name`, and `version`. These properties have no description defined in the CSV schema.
The `resource[]` properties above are currently only consumed by the web console in order to list related resources on operand details pages, but we recently had a bug opened where this API was being misused, and adding some

### PR DESCRIPTION
**Pull request by TheRealJon**
Fixes #167
Add descriptions for kind, name, and version properties of APIResourceReference type as well as DisplayName, Description, Keywords, Maintainers, Provider, Links, and Icon properties of ClusterServiceVersionSpec.

### CODE REVIEW

**Comment by TheRealJon** [comment]
I realize now after opening this that the manifests are generated files. First time contributing to an API repo like this, so I'll research a bit more and update this.

**Review by camilamacedo86 — COMMENTED** [review]
/lgtm

**Comment by TheRealJon** [comment]
@camilamacedo86 I rebased this and added a few more descriptions to the CSV API so it will need another review/lgtm

**Review by dinhxuanvu — APPROVED** [review]
/approve

**Comment by camilamacedo86** [comment]
It seems blocked by the resolution in : https://github.com/operator-framework/operator-lifecycle-manager/issues/2767
The concern is it increase the size of the CRDs and API only allows 1MB.

**Review by exdx — COMMENTED** [review]
This looks like a helpful PR, and does improve the descriptions on the CSV API, but there is an issue with the size of the CRD when applied via `kubectl apply` as the `last-applied-configuration` annotation becomes too large for the server to accept. Increasing the size of the openapi schema will further complicate this issue. 

The long term solution is to limit the depth of the codegen generated for the openapi spec, to prevent this issue from occurring. That being said I don't know whether we should prevent this PR from going in based on the issue, until the longer term fix is prepared.

File: pkg/operators/v1alpha1/clusterserviceversion_types.go
Comment (porridge) [inline pkg/operators/v1alpha1/clusterserviceversion_types.go:307]:
```suggestion
	// A list of links related to the operator.
```

File: pkg/operators/v1alpha1/clusterserviceversion_types.go
Comment (porridge) [inline pkg/operators/v1alpha1/clusterserviceversion_types.go:156]: Full stops are missing in these comments too but at least it's consistent with the comment for the containing type...

**Comment by camilamacedo86** [comment]
@msugakov, 

Your question/issue seems out of scope here, see;

>What is the point/benefit of providing k8s kinds in spec.customresourcedefinitions.owned.[*].resources at all? 
You need to inform what are the CRDs which are owned by our project so that OLM will be able to handle and install them. 
 
> In order to fix https://bugzilla.redhat.com/show_bug.cgi?id=2082491 , we need to specify that our operator manages "just some deployments"

That is not specified on this spec. If your Operator manages a deployment why are you trying to do that in the CSV? it probably should be done via the Operator implementation itself and its controllers.

**Comment by TheRealJon** [comment]
@msugakov The `spec.customresourcedefinitions.owned.[*].resources` is purely a frontend API used by the console. We use this to list resources that are managed by the operator in the topology view.

**Comment by TheRealJon** [comment]
@camilamacedo86 @exdx @porridge I've rebased this PR. Could we revisit getting this merged?

**Comment by perdasilva** [comment]
@TheRealJon you might need to do a make manifests or make verify to fix the verify stage

**Comment by tlwu2013** [comment]
looks good to me too, thanks for the PR @TheRealJon!

**Comment by tlwu2013** [comment]
+1, lgtm too, thanks for your PR @TheRealJon !

**Comment by TheRealJon** [comment]
@dinhxuanvu I brought this up in the k89s slack OLM dev channel to loop @grokspawn in. If both of you are in agreement, I think this is ready to merge.

File: pkg/operators/v1alpha1/clusterserviceversion_types.go
Comment (joelanford) [inline pkg/operators/v1alpha1/clusterserviceversion_types.go:151]: Is this `metadata.name` of the CRD? If possible, it might be nice to be a little more explicit about where these are supposed to come from if its a simple mapping from CRD fields.

Comment (TheRealJon) [reply pkg/operators/v1alpha1/clusterserviceversion_types.go:151]: Yeah, this field is the confusing one. I can update to 'metadata.name' instead of 'Name'

**Comment by grokspawn** [comment]
/hold cancel

releasing this since it looks like we had some good dialog and reached a consensus.

**Comment by joelanford** [comment]
/lgtm

Thanks!

**Pull request merged**
